src/data/pnat/factor_finsupps.lean

```diff
@@ -0,0 +1,552 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Aaron Anderson.
+ -/
+
+import data.finsupp
+import data.pnat.factors
+import order.conditionally_complete_lattice
+import order.lattice
+import order.order_iso
+import tactic.pi_instances
+
+/-!
+# Factor Finsupps
+
+A new version of prime factorisation and some facts about lattice isomorphisms
+
+## Notations
+
+## Implementation Notes
+factor_finsupp is similar to pnat.factor_multiset, but a finsupp of exponents
+Most of the sorries will be solved by proving some facts about lattice isomorphisms.
+
+## References
+
+ -/
+
+set_option old_structure_cmd true
+
+open_locale classical
+open_locale big_operators
+
+
+noncomputable theory
+
+section basics
+
+/-- Changes the domain to ℕ+. -/
```

Comment:
```suggestion
/-- Changes the domain to ℕ+. -/
```
the domain of what? Remember that these docstrings will be displayed as pop-ups when someone hovers over the name in a completely different file. So the docstrings should make sense out of context.